compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -2351,7 +2353,12 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
             // lifetime. Instead, we simply create an implicit lifetime, which will be checked
             // later, at which point a suitable error will be emitted.
             AnonymousLifetimeMode::PassThrough | AnonymousLifetimeMode::ReportError => {
-                self.new_implicit_lifetime(span)
+                if param_mode == ParamMode::Explicit {
+                    let id = self.resolver.next_node_id();
+                    self.new_named_lifetime(id, span, hir::LifetimeName::ImplicitMissing)
+                } else {
+                    self.new_implicit_lifetime(span)
+                }
```

Comment:
Probably a nicer way to do this would be to make `new_implicit_lifetime` just take a `missing: bool` arg